Source/movie.cpp

```diff
@@ -20,72 +20,63 @@ struct movie_cpp_init
 
 void __fastcall play_movie(char *pszMovie, BOOL user_can_close)
 {
-	char *v2; // esi
-	WNDPROC saveProc; // edi
-	//int v4; // eax
-	MSG Msg; // [esp+8h] [ebp-24h]
-	BOOL v6; // [esp+24h] [ebp-8h]
-	void *video_stream; // [esp+28h] [ebp-4h]
+	if ( !gbActive ) {
+		return;
+	}
 
-	v6 = user_can_close;
-	v2 = pszMovie;
-	if ( gbActive )
-	{
-		saveProc = SetWindowProc(MovieWndProc);
-		InvalidateRect(ghMainWnd, 0, 0);
-		UpdateWindow(ghMainWnd);
-		movie_playing = 1;
-		sound_disable_music(1);
-		sfx_stop();
-		effects_play_sound("Sfx\\Misc\\blank.wav");
-		SVidPlayBegin(v2, 0, 0, 0, 0, loop_movie != 0 ? 0x100C0808 : 0x10280808, &video_stream);
-		if ( video_stream )
-		{
-			do
-			{
-				if ( !gbActive || v6 && !movie_playing )
-					break;
-				while ( PeekMessage(&Msg, NULL, 0, 0, PM_REMOVE) )
-				{
-					if ( Msg.message != WM_QUIT )
-					{
-						TranslateMessage(&Msg);
-						DispatchMessage(&Msg);
-					}
+	WNDPROC saveProc = SetWindowProc(MovieWndProc);
+	InvalidateRect(ghMainWnd, 0, 0);
+	UpdateWindow(ghMainWnd);
+	movie_playing = TRUE;
+	sound_disable_music(TRUE);
+	sfx_stop();
+	effects_play_sound("Sfx\\Misc\\blank.wav");
+
+	void *video_stream;
+	SVidPlayBegin(pszMovie, 0, 0, 0, 0, loop_movie != 0 ? 0x100C0808 : 0x10280808, &video_stream);

```

Comment:
```diff
-loop_movie != 0
+!loop_movie
```

As it is a BOOL.

Would you mind updating the other two uses of `loop_movie` to `TRUE` and `FALSE`, while at it?

```
Source/monster.cpp:	loop_movie = 1;
Source/monster.cpp:	loop_movie = 0;
```

Reply:
Haha, good you also spotted my typo, it should be `loop_movie`, not `!loop_movie` as I wrote.

Reply:
Yeah it was a funny case, but off cause I also compiled and checked before submitting ;)